x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2867 @@
+
+However, since we still want to give the users the ability to update all of their rules that have updates with no conflicts in an easy and fast way, we need to introduce a distinct UI that allows for this behaviour.
+
+The current UI of the **Rule Updates** page has an button with the label "Update all" which currently simply updates all rules to their `target` version:
+
+![](2024-02-23-14-09-16.png)
+
+We can replace this button for an **Update rules with no conflicts** button. Clicking this button would make a request to `/upgrade/_perform` with a payload that includes only rules that have no conflicts. That payload can be built out of the response of the `/upgrade/_review` endpoint, including only rules with no conflicts and specifying that all rules should be updated to their `MERGED` version, which represents the "clean" (conflictless) update.
+
+```ts
+// [PSEUDOCODE]
+const reviewUpgradeResponse: ReviewRuleUpgradeResponseBody = await fetch(
+  '/internal/detection_engine/prebuilt_rules/upgrade/_review',
+  { method: 'POST' }
+);
+
+// Filter out rules with conflicts, and create payload pointing all to 'MERGE' version
+const performUpgradePayload = {
+  mode: 'SPECIFIC_RULES',
+  rules: reviewUpgradeResponse.rules
+    .filter((rule) => !rule.diff.has_conflict)
+    .map((rule) => ({
+      rule_id: rule.rule_id,
+      revision: rule.revision,
+      version: rule.version,
+    })),
+  pick_version: 'MERGE',
+};
+
+
+const performUpgradeResponse: PerformRuleUpgradeResponseBody = await fetch(
+  '/internal/detection_engine/prebuilt_rules/upgrade/_perform',
+  { method: 'POST', body: performUpgradePayload }
+);
+```
+
+#### Upgrading rules with conflicts
+
+Rules whose diffing algorithm resulted in a `CONFLICT` need to be manually resolved or confirmed by the user via the UI before making the API request to upgrade the rule.
+
+> See [Three-Way-Diff Component ticket](https://github.com/elastic/kibana/issues/171520) that details UI for solving conflicts.
+
+> See [designs](https://www.figma.com/file/gLHm8LpTtSkAUQHrkG3RHU/%5B8.7%5D-%5BRules%5D-Rule-Immutability%2FCustomization?type=design&mode=design&t=LkauhLzUKUatF6cL-0#712870904).
+
+## Other changes to UX/UI
+
+### Rules Table page
+
+- The Rules Table's filter should be updated so that rules can be fitlered according to their status as **custom**, **Elastic prebuilt** and **customized Elastic prebuilt**. Notice that **customized Elastic prebuilt** will be a subset of **Elastic prebuilt**.
+
+### Rule Details page
+
+- The Rule Details page shouls include some kind of indication or text field that lets the user know if the current rules is **custom**, **Elastic prebuilt** and **customized Elastic prebuilt**. 
+
+## Design Discussion link
+
+For an up-to-date discussion on the design changes needed for Milestone 3, please see [this ticket](https://github.com/elastic/kibana/issues/178211).
+
+
+
+
+
```

Comment:
We need to add Telemetry section - we currently collect info on the number of enabled disabled prebuilt and custom rules - potentially that one needs updates due to schema changes. @jpdjere

Reply:
That's a great point 👍 

@approksiu Also, we have https://github.com/elastic/kibana/issues/140369 for telemetry that I think we should include into Milestone 3. Could you please review and update this ticket with requirements for Milestone 3?

Reply:
- Create ticket to address changes needed to existing Telemetry for number of enabled/disabled rules, updating to reflect changes in schema (no more `immutable` field, etc)